### ISSUE TITLE
Question about extract_snps.py

### ISSUE THREAD

**Issue by stayingsong** [nservant/HiC-Pro#148]
Hi, I was trying to use HiC-Pro for allele specific analysis and had a quick question about the extract_snps.py script.

I have a VCF file for a human cell line that has been sequenced and partially phased such that it has a variety of genotype entries including 0/1, 0|1, 1/2, 1|0, 1|1, 1|2, and 2|1.

After running the extract_snps.py script on this file, I noticed that only SNPs with 1|1 entries were retained, and that in the new VCF file, they are now represented as 0/1.

My question is, why are only 1|1 (homozygous alternate) SNPs retained in the new file?  Couldn't the 0|1 and 1|0 SNPs also offer information to help map Hi-C reads to the specific alleles?

Thanks,

Michael

**Comment by stayingsong** [comment]
In fact, I am curious as to how homozygous alternate entries could even be used to phase Hi-C data, as that implies that at that position, both chromosomes have the same ALT allele, right?  So the base at that position should not be able to help one distinguish from which chromosome that read originated from right?  Whereas 0|1 and 1|0 alleles actually inform proper assignment of a base to either allele.

Or am I misunderstanding the VCF conventions or what HiC-Pro is doing?

Thanks,

Micael

**Comment by stayingsong** [comment]
Actually, now that I read the documentation over again, I just want to check if this is how HiC-Pro interprets the VCF file specified as input:

All bases corresponding to one allele are on REF, and all bases corresponding to the other allele are on ALT.  So a Hi-C read is assigned to the reference allele if all SNPs on both ends of the ditag contain REF bases.  And it is assigned to the alternate allele if all SNPs on both ends of the ditag contain ALT bases.  And the read is discarded otherwise.

Is this how it works?

Thanks for the help!

Michael

**Comment by nservant** [comment]
Hi Michael,

First of all, I would like to mention that the extractSNPs.py script has been written to extract SNPs from the Mouse Sanger database. Its use is therefore optional ... depending on your VCF file.

Then, regarding the SNPs themselves, for allele-specific analysis, you need to use the SNPs which are homozygous in parents and therefore heterozygous in F1 (your sample).
This is the answer of your first post.
So, if 0=reference allele, and 1=alternative allele, only SNPs which are 0/0 in parent 1 and 1/1 in parent 2 are used.
Why not using heterozygous SNPs ? Just because we usually does not have the exact genotype of F1 ... If parent 1 is 0/0 and parent 2 is 0/1, your F1 will therefore be either 0/1 or 0/0.
Now, let's say that you only have reads assigned to genotype 0, how can you conclude ? You will therefore say that either you have an allele-specific interaction (making the assumption that F1 is 0/1, and so that only the allele 0 has counts), or you simply has a bi-allele interaction pattern because your F1 is homozygous.
Does it make more sense now ?

Regarding HiC-Pro itself. It needs a N-masked genome where all SNPs position provided in the VCF file are replaced by a 'N'. This is your responsibility to build such reference. Reads are then aligned and this N-masked genome, and for each N, HiC-Pro looks at the nucleotide in the reads. If it finds the REF base, the read is assigned to parent 1, if it finds the ALT base, the read is assigned to the parent 2.
Any other cases, such as non expected base, or several SNPs with discordant assignment are call "ambiguous" and are discarded.
Let me know if it is still unclear.
N

**Comment by stayingsong** [comment]
Ah, thank you very much, that is a very clear and helpful explanation.

My original confusion actually came from the fact that I was working with a phased VCF file from a specific cell line derived without parental genotype information.  We actually used Hi-C data to infer the phasing information for this cell line (using HaploSeq), which is how we could assign the heterozygous SNPs which were phased to specific chromosomes.

Therefore I took the reference and alternate alleles of the phased 0|1 SNPs as is, reversed the reference and alternate alleles of the phased 1|0 SNPs, combined these two groups, then labeled them all as "0/1" prior to genome masking and mapping with HiC-Pro.  Hopefully that should work.

Thanks!

Michael
